client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -162,25 +165,50 @@ const Checkout = React.createClass( {
 	},
 
 	getCheckoutCompleteRedirectPath: function() {
-		var renewalItem,
+		var product,
+			purchasedProducts,
+			renewalItem,
+			receipt = this.props.transaction.step.data,
 			receiptId = ':receiptId';
 
 		this.props.clearPurchases();
 
 		if ( cartItems.hasRenewalItem( this.props.cart ) ) {
 			renewalItem = cartItems.getRenewalItems( this.props.cart )[ 0 ];
+			// group all purchases into an array
+			purchasedProducts = reduce( receipt && receipt.purchases || {}, function( result, value ) {
```

Comment:
If we use [_.values](https://lodash.com/docs/4.15.0#values), this could just be:

```js
purchases = values( receipt && receipt.purchases );
```